lisp/tree-sitter-cli.el

```diff
@@ -16,21 +16,49 @@
 (eval-when-compile
   (require 'subr-x))
 
-(defun tree-sitter-cli-directory ()
-  "Return tree-sitter CLI's directory, including the ending separator.
-This is the directory where the CLI tool keeps compiled lang definitions, among
-other data."
+(defvar tree-sitter-binary (executable-find "tree-sitter")
+  "Tree-sitter binary location.")
+
+(defvar tree-sitter-version (if tree-sitter-binary
+                                (nth 1 (split-string
+                                        (shell-command-to-string
+                                         (concat tree-sitter-binary " -V"))))
+                              (error "tree-sitter binary not installed"))
```

Comment:
IIRC `tree-sitter` is not a required dependency of tree-sitter (for users who are downloading precompiled shared libs), so i think we should not error here.

Reply:
style nit: it feels like there's enough logic going on here that it might be better to convert this into a function

Reply:
I've updated this now and have it returning zero so it will defaut to the `tree-sitter-cli-cache-directory`.